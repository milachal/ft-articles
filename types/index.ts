import { Dispatch, SetStateAction } from "react";

export interface HomeProps {
  articles: any;
}

export interface HomeArticleProps {
  meta: string,
  title: string
  standfirst: string,
  image: string,
}

export interface GetArticleResponse {
  meta: string,
  title: string
  standfirst: string,
  image: string,
  id: string,
}

export interface SuggestionsProps {
  suggestions: GetArticleResponse[],
  setShowSuggestions: Dispatch<SetStateAction<boolean>>
}

export interface SuggestionResult {
  apiUrl: string,
  aspectSet: string,
  id: string,
  modelVersion: string
}

export interface ArticlePageProps {
  article: {
    meta: string,
    title: string,
    standfirst: string,
    body: string,
    image: string,
    id: string
  }
}
