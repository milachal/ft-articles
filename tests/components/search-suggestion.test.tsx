import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import SearchSuggestion from "../../components/searchbar/search-suggestion";
import { SuggestionsProps } from "../../types";

const initialProps: SuggestionsProps = {
  suggestions: [{
    meta: "meta",
    title: "title",
    standfirst: "standfirst",
    body: "body",
    image: "url",
    id: "/thing/id"
  }],
  setShowSuggestions: jest.fn(),
  setQuery: jest.fn()
};

jest.mock("../../styles/search-suggestion.scss", () => <div />);

describe("SearchSuggestion component", () => {
  const wrapper: ShallowWrapper = shallow(<SearchSuggestion {...initialProps} />);

  it("tests if components is rendered correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should test component content", () => {
    const title = wrapper.find("[data-testid=\"title\"]");
    const image = wrapper.find("[data-testid=\"image\"]");

    expect(title.exists()).toBe(true);
    expect(title.text()).toBe("title");
    expect(image.exists()).toBe(true);
    expect(image.prop("src")).toEqual("url");
  });
});
