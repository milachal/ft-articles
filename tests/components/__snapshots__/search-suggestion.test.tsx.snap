// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSuggestion component tests if components is rendered correctly 1`] = `
<div
  className="suggestions-container"
>
  <div
    className="suggestions-container__suggestion"
    key="id"
  >
    <Link
      href="/article/id"
    >
      <a>
        <div
          className="suggestions-container__title"
          data-testid="title"
          onClick={[Function]}
        >
          title
        </div>
      </a>
    </Link>
    <div
      className="suggestions-container__image-wrapper"
    >
      <img
        alt="article"
        data-testid="image"
        src="url"
      />
    </div>
  </div>
</div>
`;
